src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/ContextQuery/VisualBasicSyntaxContext.vb

```diff
@@ -170,6 +170,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
 
         Friend Overrides Function IsAwaitKeywordContext() As Boolean
             If IsAnyExpressionContext OrElse IsSingleLineStatementContext Then
+                If IsInQuery Then
+                    ' There actually are some places where Await is allowed, but we don't support these. See BC36929 for details:
+                    ' BC36929: 'Await' may only be used in a query expression within the first collection expression of the initial 'From' clause or within the collection expression of a 'Join' clause.
+                    Return False
```

Comment:
i would not want this regression. if we allow this in the from-expression, then we should allow this :)

Reply:
Working on it. There is also
`error CS1995: The 'await' operator may only be used in a query expression within the first collection expression of the initial 'from' clause or within the collection expression of a 'join' clause` for C#.

Reply:
Done in [d94e9de..0d892bf](https://github.com/dotnet/roslyn/pull/55817/files/d94e9decf31df66d1fbd72fe11062bd3ac12af95..0d892bfe71ee3a92347c137666b2f87b1333a38b)